README.md

```diff
@@ -1,2 +1,125 @@
-# nbde_client
-Ansible role for configuring Network Bound Disk Encryption clients (e.g. clevis)
+nbde_client
+-----------
+
+Ansible role for configuring Network-Bound Disk Encryption clients (e.g. clevis).
+
+This role currently supports `clevis` as a provider and it uses it for operations like encryption
+and decryption.
+
+
+Supported Distributions
+-----------------------
+* RHEL-7+, CentOS-7+
+* Fedora
+
+
+Limitations
+-----------
+This role can currently create `tang` bindings. TPM2 is not supported as of now.
+
+
+Role Variables
+--------------
+
+These are the variables that can be passed to the role:
+
+
+| **Variable** | **Default/Choices** | **Description** |
+|----------|-------------|------|
+| `nbde_client_provider` | `clevis`| identifies the provider for the `nbde_client` role. We currently support `clevis`.|
+| `nbde_client_bindings` | | a list containing groups of LUKS devices and applicable settings for a binding. This group of devices contains specific binding configurations to be applied to the set of devices.
+|`nbde_client_remote_dir` | `/root/.local/share/nbde_client/` |  specifies a directory in the remote hosts that may be used for storing temporary data such as transferred keyfiles. |
+|`nbde_client_update_initramfs` | `yes` | indicates whether the initramfs should be updated in case changes are made when processing the clevis operations on the devices.|
+
+
+#### nbde_client_bindings
+`nbde_client_bindings` supports the following keys:
+| **Name** | **Default/Choices** | **Description** |
+|----------|-------------|------|
+|`devices` | | specifies LUKS devices, with their path and possibly either a passphrase or key file.|
+|`auth` | | the settings for the binding, like the servers it should bind to. |
+| `state` | **present** / absent | specifies whether the binding with the configuration described should be added or removed. Setting `state` to `present` (the default) means bindings will be added; setting `state` to `absent` means bindings will be removed from the devices/slots. |
```

Comment:
What is the primary key to identify the binding? Is it `auth.slot`  ? it seems so, as the example below says:

#### Example 2: remove binding from slot 2
```yaml
---
- hosts: all
  vars:
    nbde_client_bindings:
      - devices:
          - path: /dev/sda1
            pass: passphrase
        auth:
          slot: 2
        state: absent
```

Reply:
Or is it the combination of (device, slot) ?

Reply:
Yes, it's device and slot. I guess it makes sense to move slot out from auth.

Reply:
I think so, yes. Shouldn't the slot be per-device? if one binding is using two device, can the slot be different for the two? (I am really unfamiliar with nbde concepts.)

Reply:
Each device may have bindings (more than one, even), and each binding is associated with a slot. So yeah, moving the slot to device makes sense here. We could have different devices setting the bindings in different slots.

Reply:
What happens if I run the role and add a binding with two devices and then I run the role again and I specify just one of the devices and say `state: absent`? Will the binding be split into two bindings and the one that corresponds to the device given will be removed, while the other will stay?

Reply:
If you have two devices in there, one binding will be added in each device, in a specific slot. 
/dev/sda1 and /dev/sda2, for instance. You will have a binding in each of them, meaning that you will be able to unlock each of them (with eg. clevis), via the existing bindings.

If you afterwards specify /dev/sda2 with state absent, you will remove the binding existing in the specified device; you will not be able to unlock this one device, unless you have other bindings in other LUKS slots.

Reply:
But then it is not one binding with two devices, it is two bindings, each with one device. So the way you are representing it in the interface is inaccurate.

Reply:
I agree. It seemed simpler to use with multiple devices, but in the end it became confusing and not as accurate.
What about something like this, instead, to describe the bindings?

```
{
	device
	pass
	keyfile
	slot
	state
	discard_passphrase
	auth {
		threshold
		servers
	}
}
```

Reply:
looks good, maybe it is not needed to group `threshold` with `servers` under `auth` and you could add those two directly in the dict?

Reply:
Yeah, I think it would make things simpler that way. Let me update this.

Reply:
Updated.